### PR TITLE
plan, tidb-server: add a flag to control statistic info reading.

### PR DESCRIPTION
**Pull request by hanfei1991**
@shenli @coocood @zimulala @tiancaiamao @lamxTyler PTAL

### CODE REVIEW

File: plan/logical_plan_builder.go [hunk 1]
```diff
@@ -912,7 +913,10 @@ func (b *planBuilder) buildTableDual() LogicalPlan {
 }
 
 func (b *planBuilder) buildDataSource(tn *ast.TableName) LogicalPlan {
-	statisticTable := statscache.GetStatisticsTableCache(b.ctx, tn.TableInfo)
+	statisticTable := statistics.PseudoTable(tn.TableInfo)
```
Comment (shenli) [inline plan/logical_plan_builder.go]: Can we move this after the following if?

**Comment by shenli** [comment]
CI failed.

File: plan/optimizer.go [hunk 2]
```diff
@@ -27,6 +27,9 @@ import (
 // AllowCartesianProduct means whether tidb allows cartesian join without equal conditions.
 var AllowCartesianProduct = true
 
+// EnableStatistic means tidb will use the statistic information to do cost-based information.
```
Comment (lamxTyler) [inline plan/optimizer.go]: `EnableStatistic means whether tidb uses the statistic information to do cost-based optimization` is more uniform with `AllowCartesianProduct`.

**Comment by lamxTyler** [comment]
It seems this flag is only used to control reading, will it be used to control `analyze` statement？ After all if we do not use stats info in optimizer, there is no need to analyze it.

**Comment by hanfei1991** [comment]
@lamxTyler `analyze` will store info to tikv, may be used by other tidb

**Comment by lamxTyler** [comment]
LGTM

**Pull request merged**